[BE][BUG] LocalDateTime 역직렬화 문제 해결

## 🤷 버그 내용
로깅 시 직렬화된 LocalDateTime을 String 형식으로 역직렬화가 안되기 때문에
reservation을 생성할때 다음과 같은 오류가 발생합니다.
```
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.LocalDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: org.example.backend.reservation.domain.dto.ReservationResDto["startTime"])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1330) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:502) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4799) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4040) ~[jackson-databind-2.17.2.jar!/:2.17.2]
	at org.example.backend.common.aop.logging.LoggingAspect.logRequestResponse(LoggingAspect.java:65) ~[!/:0.0.1-SNAPSHOT]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:637) ~[spring-aop-6.1.12.jar!/:6.1.12]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:627) ~[spring-aop-6.1.12.jar!/:6.1.12]
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71) ~[spring-aop-6.1.12.jar!/:6.1.12]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.1.12.jar!/:6.1.12]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768) ~[spring-aop-6.1.12.jar!/:6.1.12]
	... 106 common frames omitted
```

## ⚠ 버그 재현 방법
1. 로그인한다.
2. 예약을 생성한다.

## 📸 스크린샷

<img width="700" alt="Image" src="https://github.com/user-attachments/assets/51e291e2-f4d8-45b1-bf52-1277901e0abe" />

## 👄 참고 사항

--- a/backend/src/main/java/org/example/backend/common/aop/logging/LoggingAspect.java
+++ b/backend/src/main/java/org/example/backend/common/aop/logging/LoggingAspect.java
@@ -2,6 +2,7 @@ package org.example.backend.common.aop.logging;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import jakarta.servlet.http.HttpServletRequest;
 import java.time.LocalDateTime;
 import java.util.Arrays;
@@ -29,7 +30,7 @@ public class LoggingAspect {
 
     private static final List<String> EXCLUDE_NAMES = Arrays.asList("fileList", "request");
     private static final int MAX_RESPONSE_BODY_LENGTH = 65000;
-    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
     private final Logger log = LoggerFactory.getLogger(this.getClass());
 
     private final BulkLogManager bulkLogManager;

--- a/backend/src/main/java/org/example/backend/reservation/repository/ReservationRepository.java
+++ b/backend/src/main/java/org/example/backend/reservation/repository/ReservationRepository.java
@@ -10,8 +10,8 @@ import org.springframework.data.repository.query.Param;
 public interface ReservationRepository extends JpaRepository<Reservation, Long> {
     @Query("SELECT COUNT(r) > 0 FROM Reservation r " +
             "WHERE r.room.id = :seminarRoomId " +
-            "AND r.startTime <= :endTime " +
-            "AND r.endTime >= :startTime")
+            "AND r.startTime < :endTime " +
+            "AND r.endTime > :startTime")
     boolean existsByTimePeriod(
             @Param("seminarRoomId") Long seminarRoomId,
             @Param("startTime") LocalDateTime startTime,

